swap/swap.go

```diff
@@ -80,8 +85,41 @@ func NewParams() *Params {
 	}
 }
 
+// newLogger returns a new logger
+func newLogger(logpath string) log.Logger {
+	swapLogger := log.New("swaplog", "*")
+
+	defaultHandler := log.Root().GetHandler()
+	rfh, err := swapRotatingLogHandler(logpath)
+
+	if err != nil {
+		//default handler just in case the handler is not set
+		log.Warn("RotatingFileHandler was not initialized", "logdir", logpath, "err", err)
+		swapLogger.SetHandler(defaultHandler)
+		return swapLogger
+	}
+
+	//Filters messages with the correct logLevel for swap
+	rfh = log.LvlFilterHandler(log.Lvl(swapLogLevel), rfh)
+
+	//Dispatches the logs to the default swarm log and also the filtered swap file logger.
+	swapLogger.SetHandler(log.MultiHandler(defaultHandler, rfh))
+
+	return swapLogger
+}
+
+// swapRotatingLogHandler returns a RotatingFileHandler or Swarm Logger
```

Comment:
shouldn't this be called something like `swapRotatingFileHandler` or `swapLogRotatingHandler` instead?

Reply:
fixed